src/AppInstallerCommonCore/Registry.cpp

```diff
@@ -439,6 +439,79 @@ namespace AppInstaller::Registry
         return result;
     }
 
+    Key Key::CreateKeyAndOpen(HKEY key, std::string_view subKey, DWORD options, REGSAM access)
+    {
+        return CreateKeyAndOpen(key, Utility::ConvertToUTF16(subKey), options, access);
+    }
+
+    Key Key::CreateKeyAndOpen(HKEY key, const std::wstring& subKey, DWORD options, REGSAM access)
+    {
+        Key result;
+        result.Create(key, subKey, options, access);
+        return result;
+    }
+
+    bool Key::DeleteKey(HKEY key, std::string_view subKey, DWORD samDesired)
+    {
+        return DeleteKey(key, Utility::ConvertToUTF16(subKey), samDesired);
+    }
+
+    bool Key::DeleteKey(HKEY key, const std::wstring& subKey, DWORD samDesired)
+    {
+        LSTATUS status = RegDeleteKeyExW(key, subKey.c_str(), samDesired, 0);
+        if (status == ERROR_SUCCESS)
+        {
+            AICLI_LOG(Core, Verbose, << "Subkey '" << Utility::ConvertToUTF8(subKey) << "' was deleted successfully.");
+            return true;
+        }
+
+        THROW_IF_WIN32_ERROR(status);

```

Comment:
You are throwing on error, so there is no need to have a `bool` return value.  I could see keeping one and turning the "path not found" error into false though.